File: app/src/main/java/com/worldventures/dreamtrips/wallet/analytics/oncard/action/SmartCardAddRecordAction.java
package com.worldventures.dreamtrips.wallet.analytics.oncard.action;

import com.worldventures.core.service.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.AdobeTracker;

import io.techery.janet.smartcard.model.analytics.AnalyticsLog;

@AnalyticsEvent(action = "wallet:oncard:card add", trackers = AdobeTracker.TRACKER_KEY)
class SmartCardAddRecordAction extends SmartCardAnalyticsAction {

   SmartCardAddRecordAction(AnalyticsLog logEntry) {
      super(logEntry);
   }

   @Override
   protected void processLog(int type, AnalyticsLog logEntry) {
      super.processLog(type, logEntry);
      switch (type) {
         case AnalyticsLog.CARD_READ_PASS:
            attributeMap.put("ocsuccardswipe", "1");
            break;
         case AnalyticsLog.CARD_READ_ERROR:
            attributeMap.put("ocfailcardswipe", "1");
            break;
         case AnalyticsLog.CARD_READ_FORMAT_ERROR:
            attributeMap.put("ocformaterrorswipe", "1");
            break;
         case AnalyticsLog.CARD_READ_NAME_ERROR:
            attributeMap.put("ocnameerrorswipe", "1");
            break;
         default:
            break;
      }
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/oncard/action/SmartCardAddRecordAction.kt
package com.worldventures.wallet.analytics.oncard.action

import com.worldventures.janet.analytics.AnalyticsEvent
import com.worldventures.core.service.analytics.AdobeTracker

import io.techery.janet.smartcard.model.analytics.AnalyticsLog

@AnalyticsEvent(action = "wallet:oncard:card add", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
internal class SmartCardAddRecordAction(logEntry: AnalyticsLog) : SmartCardAnalyticsAction(logEntry) {

   override fun processLog(type: Int, logEntry: AnalyticsLog) {
      super.processLog(type, logEntry)
      when (type) {
         AnalyticsLog.CARD_READ_PASS -> attributeMap.put("ocsuccardswipe", "1")
         AnalyticsLog.CARD_READ_ERROR -> attributeMap.put("ocfailcardswipe", "1")
         AnalyticsLog.CARD_READ_FORMAT_ERROR -> attributeMap.put("ocformaterrorswipe", "1")
         AnalyticsLog.CARD_READ_NAME_ERROR -> attributeMap.put("ocnameerrorswipe", "1")
      }
   }
}